### PR TITLE
Feature/exclude return result

### PR DESCRIPTION
**Pull request by supermacro**
The `must-use` rule must allow for defining functions that return `Result`s.

## Changes:

- Formatting in test file
- Addition of new test case to ensure that a user can define functions that return `Result`
  - This test is currently failing as I have not implemented the logic inside of `must-use-result.ts`
- Addition of `yarn.lock` file

## Todo: 

- Implement logic to allow test to pass
  - Implementing eslint rules is not something I am familiar with ... as such we have two options: Allow me some time to learn how to actually do this ... or simply push the solution to this branch.

### CODE REVIEW

**Comment by mdbetancourt** [comment]
oh actually i implemented it but i commented that line while was debugging

File: src/rules/must-use-result.ts [hunk 1]
```diff
@@ -25,91 +29,110 @@ const resultProperties = [
   'orElse',
   'orElse',
   'match',
```
Comment (supermacro) [inline src/rules/must-use-result.ts:31]: This is probably a dumb question 🙈 but i'm wondering why there's a need to specify duplicate property names? 

eg `angThen` and `orElse` being repeated 3 times each

File: src/rules/must-use-result.ts [hunk 2]
```diff
@@ -25,91 +29,110 @@ const resultProperties = [
...
-  'unwrapOr'
-]
+  'unwrapOr',
+];
 
-const handledMethods = [
-  'match',
-  'unwrapOr'
-]
+const handledMethods = ['match', 'unwrapOr'];
```
Comment (supermacro) [inline src/rules/must-use-result.ts:35]: Shouldn't `_unsafeUnwrap` also be part of this list?

**Comment by supermacro** [comment]
FYI @mdbetancourt posted 2 questions :)

File: src/rules/must-use-result.ts (hunk 2 above)
Comment (mdbetancourt) [reply src/rules/must-use-result.ts:35]: good question  🤔, fixed

**Pull request merged**